components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManagerImpl.java

```diff
@@ -0,0 +1,640 @@
+
+    /**
+     * {@inheritDoc}
+     */
+    public Resources getResources() throws ConfigurationManagementException {
+
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public Resources getResourcesByType(String resourceType) throws ConfigurationManagementException {
+
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public Resource getResource(String resourceTypeName, String resourceName)
+            throws ConfigurationManagementException {
+
+        validateResourceRetrieveRequest(resourceTypeName, resourceName);
+        ResourceType resourceType = getResourceType(resourceTypeName);
+        Resource resource = this.getConfigurationDAO()
+                .getResourceByName(getTenantId(), resourceType.getId(), resourceName);
+        if (resource == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("No resource found for the resourceName: " + resourceName);
+            }
+            throw ConfigurationUtils.handleClientException(
+                    ErrorMessages.ERROR_CODE_RESOURCE_DOES_NOT_EXISTS, resourceName, null);
+        }
+        return resource;
+    }
+
+    private void validateResourceRetrieveRequest(String resourceTypeName, String resourceName)
+            throws ConfigurationManagementException {
+
+        if (StringUtils.isEmpty(resourceName) || StringUtils.isEmpty(resourceTypeName)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Invalid resource identifier with resourceName: " + resourceName
+                        + " and resourceTypeName: " + resourceTypeName + ".");
+            }
+            throw handleClientException(ERROR_CODE_RESOURCE_GET_REQUEST_INVALID, null);
+        }
+    }
+
+    private int getTenantId() {
+
+        return PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+    }
+
+    private String getTenantDomain() {
+
+        return PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void deleteResource(String resourceTypeName, String resourceName) throws ConfigurationManagementException {
+
+        validateResourceDeleteRequest(resourceTypeName, resourceName);
+        ResourceType resourceType = getResourceType(resourceTypeName);
+        this.getConfigurationDAO().deleteResourceByName(getTenantId(), resourceType.getId(), resourceName);
+        if (log.isDebugEnabled()) {
+            log.debug(StringUtils.capitalize(resourceName) + " configuration deleted successfully.");
+        }
+    }
+
+    private void validateResourceDeleteRequest(String resourceTypeName, String resourceName)
+            throws ConfigurationManagementException {
+
+        if (StringUtils.isEmpty(resourceName) || StringUtils.isEmpty(resourceTypeName)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error identifying the resource with resource name: " + resourceName + " and resource type:"
```

Comment:
If this is an error case we can make the it a error log instead

Reply:
No need to make it as error.. since it is a client error.